demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -0,0 +1,625 @@
+                                         const char * pHost,
+                                         const char * pMethod,
+                                         const char * pPath )
+{
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* The location of the file size in contentRangeValStr. */
+    char * pFileSizeStr = NULL;
+
+    /* String to store the Content-Range header value. */
+    char * contentRangeValStr = NULL;
+    size_t contentRangeValStrLength = 0;
+
+    /* Initialize the request object. */
+    requestInfo.pHost = pHost;
+    requestInfo.hostLen = strlen( pHost );
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = strlen( pMethod );
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. This is done in
+     * order to download the file in parts. */
+    requestInfo.reqFlags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    /* Initialize the response object. The same buffer used for storing
+     * request headers is reused here. */
+    response.pBuffer = userBuffer;
+    response.bufferLen = USER_BUFFER_LENGTH;
+
+    LogInfo( ( "Getting file object size from host..." ) );
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Add the header to get bytes=0-0. S3 will respond with a Content-Range
+         * header that contains the size of the file in it. This header will look
+         * like: "Content-Range: bytes 0-0/FILESIZE". The body will have a single
+         * byte that we are ignoring. */
+        httpStatus = HTTPClient_AddRangeHeader( &requestHeaders, 0, 0 );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      NULL,
+                                      0,
+                                      &response,
+                                      0 );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
+                   pHost, pPath ) );
+        LogInfo( ( "Response Headers:\n%.*s",
+                   ( int32_t ) response.headersLen,
+                   response.pHeaders ) );
+        LogInfo( ( "Response Status:\n%u",
+                   response.statusCode ) );
+        LogInfo( ( "Response Body:\n%.*s\n",
+                   ( int32_t ) response.bodyLen,
+                   response.pBody ) );
+
+        httpStatus = HTTPClient_ReadHeader( &response,
```

Comment:
Same comment as https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1318

https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1318/files#r508773748

Please check for a 206 status-code in response.statusCode before proceeding.